ceph/rados/core_workflows.py

```diff
@@ -271,6 +271,97 @@ def get_pool_property(self, pool, props):
         prop_details = json.loads(out)
         return prop_details
 
+    def host_maintenance_enter(self, hostname: str, retry: int = 10) -> bool:
+        """
+        Adds the specified host into maintenance mode
+        Args:
+            hostname: name of the host which needs to be added into maintenance mode
+            retry: Retry the command to add host into maintainence mode
+        Returns:
+            True -> Host successfully added to maintenance mode
+            False -> Host Could not be added to maintenance mode
+        """
+        log.debug(f"Passed host : {hostname} to be added into maintenance mode")
+        iteration = 0
+
+        while iteration <= retry:
+            iteration += 1
+            cmd = f"ceph orch host maintenance enter {hostname} --force"
+            out, _ = self.client.exec_command(cmd=cmd, sudo=True, timeout=600)
+            log.debug(f"o/p of maintenance enter cmd : {out}")
+            time.sleep(20)
+
+            # Checking the status of the host in ceph orch host ls, if the status has been marked as maintenance mode
+            cmd = "ceph orch host ls"
+            out = self.run_ceph_command(cmd=cmd, client_exec=True)
+
+            flag = False
+            for entry in out:
+                if entry["hostname"].lower() == hostname.lower():
+                    flag = True
```

Comment:
[minor] we can also use the common module to fetch the status for the particular host directly- 
```
ceph.rados.core_workflows.RadosOrchestrator
def check_host_status(self, hostname: Any) -> bool
```

Reply:
Made necessary changes